internal/provisioner/plain/controllers/bundledeployment_controller.go

```diff
@@ -132,7 +132,11 @@ func (r *BundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		if bundle.Status.Phase == rukpakv1alpha1.PhaseFailing {
 			reason = rukpakv1alpha1.ReasonUnpackFailed
 			status = metav1.ConditionFalse
-			message = fmt.Sprintf("Failed to unpack the %s Bundle", bundle.GetName())
+			bundleMessage := "No unpacked condition"
+			if c := meta.FindStatusCondition(bundle.Status.Conditions, rukpakv1alpha1.TypeUnpacked); c != nil {
+				bundleMessage = c.Message
+			}
+			message = fmt.Sprintf("Failed to unpack the %s Bundle: %s", bundle.GetName(), bundleMessage)
```

Comment:
I think we're almost there. Given we know where handling a failing Bundle unpack phase, tacking on a "No unpacked condition" might not be the greatest UX here.

```
message := fmt.Sprintf("Failed to unpack the %s Bundle", bundle.GetName())
if c := meta.FindStatusCondition(bundle.Status.Conditions, rukpakv1alpha1.TypeUnpacked); c != nil {
    message = fmt.Sprintf("%s: %s", message, c.Message)
}
...
```

We should just be able to override the default message variable and append that condition's message after the original value.

Reply:
I see.  That's better.

Reply:
It's clean 3 line change now :-)